app/controllers/games_controller.rb

```diff
@@ -13,17 +13,17 @@ def create
 
   def show
     @game = Game.find_by_id(params[:id])
-
     data = params[:p_type]
-
+    return render text: 'Not Found', status: :not_found if @game.blank?
     until data.nil?
       @game.pieces.where(type: 'Pawn', vertical_position: 1).first.try(:update_attributes, type: data)
       @game.pieces.where(type: 'Pawn', vertical_position: 8).first.try(:update_attributes, type: data)
       data = params[:p_type] = nil
-      redirect_to game_path(@game)
+      return redirect_to game_path(@game)
```

Comment:
@trustinruss and I paired on this last night. We found that not having the return there somehow broke the test for pawn promotion in the game controller spec.